"use strict";(self.webpackChunk_N_E=self.webpackChunk_N_E||[]).push([[1410],{1410:(e,t,n)=>{n.r(t),n.d(t,{default:()=>O});var a=n(12115),r=n(80770),i=n.n(r),o=n(20109),l=n(67072),s=n(97717),c=n.n(s),d=n(96256),m=n.n(d),u=n(88153),h=n(79201),p=n(17386);let f=(0,n(97771).makeStyles)()({compact:{paddingRight:0,paddingTop:0,paddingBottom:0}}),g=["read paired","read mapped in proper pair","read unmapped","mate unmapped","read reverse strand","mate reverse strand","first in pair","second in pair","not primary alignment","read fails platform/vendor quality checks","read is PCR or optical duplicate","supplementary alignment"];function b(e){let{classes:t}=f(),{feature:n}=e,{flags:r}=n;return a.createElement(c(),{...e,title:"Flags"},a.createElement(m(),{name:"Flag",value:r}),a.createElement(u.A,null,g.map((e,n)=>{let i=r&1<<n,o=`${e}_${i}`;return a.createElement(h.A,{key:o,control:a.createElement(p.A,{className:t.compact,checked:!!i,name:e,readOnly:!0}),label:e})})))}var y=n(21079),C=n.n(y);function v({value:e}){let[t,n]=(0,a.useState)(!1),[r,i]=(0,a.useState)(!1),o=String(e);return o.length>100?a.createElement(a.Fragment,null,a.createElement("button",{type:"button",onClick:()=>{C()(o),i(!0),setTimeout(()=>{i(!1)},700)}},r?"Copied to clipboard":"Copy"),a.createElement("button",{type:"button",onClick:()=>{n(e=>!e)}},t?"Show less":"Show more"),a.createElement("div",null,t?o:`${o.slice(0,100)}...`)):a.createElement("div",null,o)}var q=n(28254),E=n(13516);function w({locString:e,model:t}){return a.createElement(q.A,{onClick:n=>{n.preventDefault(),(0,E.h)(e,t)},href:"#"},e)}let S={AM:"The smallest template-independent mapping quality in the template",AS:"Alignment score generated by aligner",BC:"Barcode sequence identifying the sample",BQ:"Offset to base alignment quality (BAQ)",BZ:"Phred quality of the unique molecular barcode bases in the {OX} tag",CB:"Cell identifier",CC:"Reference name of the next hit",CM:"Edit distance between the color sequence and the color reference (see also {NM})",CO:"Free-text comments",CP:"Leftmost coordinate of the next hit",CQ:"Color read base qualities",CR:"Cellular barcode sequence bases (uncorrected)",CS:"Color read sequence",CT:"Complete read annotation tag, used for consensus annotation dummy features",CY:"Phred quality of the cellular barcode sequence in the {CR} tag",E2:"The 2nd most likely base calls",FI:"The index of segment in the template",FS:"Segment suffix",FZ:"Flow stopToken intensities",GC:"Reserved for backwards compatibility reasons",GQ:"Reserved for backwards compatibility reasons",GS:"Reserved for backwards compatibility reasons",H0:"Number of perfect hits",H1:"Number of 1-difference hits (see also {NM})",H2:"Number of 2-difference hits",HI:"Query hit index",IH:"Query hit total count",LB:"Library",MC:"CIGAR string for mate/next segment",MD:"String encoding mismatched and deleted reference bases",MF:"Reserved for backwards compatibility reasons",MI:"Molecular identifier; a string that uniquely identifies the molecule from which the record was derived",ML:"Base modification probabilities",MM:"Base modifications / methylation ",MQ:"Mapping quality of the mate/next segment",NH:"Number of reported alignments that contain the query in the current record",NM:"Edit distance to the reference",OA:"Original alignment",OC:"Original CIGAR (deprecated; use {OA} instead)",OP:"Original mapping position (deprecated; use {OA} instead)",OQ:"Original base quality",OX:"Original unique molecular barcode bases",PG:"Program",PQ:"Phred likelihood of the template",PT:"Read annotations for parts of the padded read sequence",PU:"Platform unit",Q2:"Phred quality of the mate/next segment sequence in the {R2} tag",QT:"Phred quality of the sample barcode sequence in the {BC} tag",QX:"Quality score of the unique molecular identifier in the {RX} tag",R2:"Sequence of the mate/next segment in the template",RG:"Read group",RT:"Reserved for backwards compatibility reasons",RX:"Sequence bases of the (possibly corrected) unique molecular identifier",S2:"Reserved for backwards compatibility reasons",SA:"Other canonical alignments in a chimeric alignment",SM:"Template-independent mapping quality",SQ:"Reserved for backwards compatibility reasons",TC:"The number of segments in the template",TS:"Transcript strand",U2:"Phred probability of the 2nd call being wrong conditional on the best being wrong",UQ:"Phred likelihood of the segment, conditional on the mapping being correct"},k=(0,a.lazy)(()=>n.e(5129).then(n.bind(n,25129))),R=(0,a.lazy)(()=>n.e(4844).then(n.bind(n,84844))),O=(0,l.observer)(function(e){let{model:t}=e,{featureData:n}=t,r=structuredClone(n),l=(0,E.b)("SA",r),{flags:s}=r;return a.createElement(o.A,{"data-testid":"alignment-side-drawer"},a.createElement(i(),{...e,descriptions:{tags:S},feature:r,formatter:(e,n)=>"next_segment_position"===n?a.createElement(w,{model:t,locString:e}):a.createElement(v,{value:e})}),void 0!==l?a.createElement(k,{model:t,tag:l,feature:r}):null,1&s?a.createElement(R,{model:t,feature:r}):null,void 0!==s?a.createElement(b,{feature:r,...e}):null)})},13516:(e,t,n)=>{n.d(t,{b:()=>r,h:()=>i});var a=n(54262);function r(e,t){var n;return(null===(n=t.tags)||void 0===n?void 0:n[e])||t[e]}async function i(e,t){let n=(0,a.getSession)(t),{view:r}=t;try{if(r)await r.navToLocString(e);else throw Error("No view associated with this view anymore")}catch(e){console.error(e),n.notify(`${e}`)}}}}]);
//# sourceMappingURL=1410.95a3ba1bd5b412d8.js.map